url47087446e9_kellrott_SparqlScape_tgz-pJ8-org_json_TestJ8/src/org/sparqlscape/NameSpaceEditor.java

```diff
@@ -0,0 +1,102 @@
+package org.sparqlscape;
+
+import java.awt.Button;
+import java.awt.Dimension;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.table.DefaultTableModel;
+
+public class NameSpaceEditor {
+
+	public static Map<String,String> EditDialog( Map<String,String> props ) {
+		final JDialog frame = new JDialog();
+		final Map<String,String> out = new HashMap<String,String>(props);
+		JPanel panel = new JPanel();
+		panel.setLayout( new GridBagLayout() );
+		GridBagConstraints c = new GridBagConstraints();
+		c.gridy = 0;
+		c.gridx = 0;
+		c.gridwidth = 2;
+		//Edge List
+		final DefaultTableModel tableModel = new DefaultTableModel(props.size(), 2);		
+		int i = 0;
+		for ( String key : props.keySet() ) {
+			tableModel.setValueAt(key, i, 0);
+			tableModel.setValueAt(props.get(key) , i, 1);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.